### ISSUE TITLE
Handling uncaught exception in showCheckout()

### ISSUE THREAD

**Issue by rgbworld** [PaddleHQ/Mac-Framework-V4#6]
When using `paddle.showCheckout()`, the debugger is logging "__Handling uncaught exception__", and checkout fails. In the Paddle checkout sheet, the message "Completing Purchase" is displayed with the progress spinner and it just hangs there. The order is placed and emails are sent, however, the Paddle checkout window must be closed via the Cancel button because it just sits there and spins. The completion block is never called on success, only when cancelled.

Note: My implementation uses DevMateKit for licensing & activation. Paddle is used exclusively for checkout. Previously (with Paddle.framework v3), I was using `paddle.purchaseProduct(productId, with: window, completionBlock: completionBlock)`, so it appears now I must use `paddle.showCheckout()`.

Certainly could be my mistake, but I spent several hours trying for successful checkout without success. My product uses a subscription license via DevMateKit.

Example usage (as a sheet):

    func paddlePurchaseProduct(productId: Int, inWindow window: NSWindow?) {
        guard let paddle = Paddle.sharedInstance() else {
            return
        }

        let paddleProduct = PADProduct(productID: "\(productId)", productType: PADProductType.product, configuration: defaultProductConfig())!
        let completionBlock = self.statusCompletionBlock()
        paddle.showCheckout(for: paddleProduct, options: nil, checkoutStatusCompletion: completionBlock)
    }

    func statusCompletionBlock() -> PADCheckoutStateCompletion {
        return { (state, data) in
            // never called with .purchased state
        }
    }

    func defaultProductConfig() -> PADProductConfiguration {
        let defaultProductConfig = PADProductConfiguration()
        defaultProductConfig.productName = "My Product"
        defaultProductConfig.vendorName = "My Company"
        return defaultProductConfig
    }

    func willShowPaddle(_ uiType: PADUIType, product: PADProduct) -> PADDisplayConfiguration? {
        let parentWindow = DMActivationController.shared().window // example parent window
        let dispConfig = PADDisplayConfiguration.init(.sheet, hideNavigationButtons: true, parentWindow: parentWindow)
        return dispConfig
    }

## Report

```
__exceptionPreprocess, '*** -[__NSArray0 objectAtIndex:]: index 0 beyond bounds for empty NSArray'

0   com.apple.CoreFoundation            0x00007fff336612c3 __exceptionPreprocess + 147
1   libobjc.A.dylib                     0x00007fff5a802c76 objc_exception_throw + 48
2   com.apple.CoreFoundation            0x00007fff3367b8cd -[__NSArray0 objectAtIndex:] + 93
3   com.paddle.Paddle                   0x000000010293ca26 -[PADCheckoutApiOrderStatus firstLicenseCode] + 169
4   com.paddle.Paddle                   0x0000000102933792 __57-[PADCheckoutWindowController completeCheckout:andEmail:]_block_invoke + 244
5   com.paddle.Paddle                   0x0000000102937843 __40+[PADOrderStatusPoller poll:completion:]_block_invoke_2 + 188
6   com.paddle.Paddle                   0x00000001029362bb __48-[PADCheckoutApi requestOrderStatus:completion:]_block_invoke + 113
7   com.paddle.Paddle                   0x0000000102935e5d __61-[PADCheckoutApi call:method:unencodedParameters:completion:]_block_invoke + 336
8   com.apple.CFNetwork                 0x00007fff3250484b __75-[__NSURLSessionLocal taskForClass:request:uploadFile:bodyData:completion:]_block_invoke + 19
9   com.apple.CFNetwork                 0x00007fff325040a5 __49-[__NSCFLocalSessionTask _task_onqueue_didFinish]_block_invoke + 147
10  com.apple.Foundation                0x00007fff356f35df __NSBLOCKOPERATION_IS_CALLING_OUT_TO_A_BLOCK__ + 7
11  com.apple.Foundation                0x00007fff356f3441 -[NSBlockOperation main] + 68
12  com.apple.Foundation                0x00007fff356f18ee -[__NSOperationInternal _start:] + 778
13  libdispatch.dylib                   0x00007fff5b3e2db8 _dispatch_client_callout + 8
14  libdispatch.dylib                   0x00007fff5b3f533e _dispatch_block_invoke_direct + 317
15  libdispatch.dylib                   0x00007fff5b3e2db8 _dispatch_client_callout + 8
16  libdispatch.dylib                   0x00007fff5b3f533e _dispatch_block_invoke_direct + 317
17  libdispatch.dylib                   0x00007fff5b3f51e1 dispatch_block_perform + 109
18  com.apple.Foundation                0x00007fff356ed8fc __NSOQSchedule_f + 342
19  libdispatch.dylib                   0x00007fff5b3e2db8 _dispatch_client_callout + 8
20  libdispatch.dylib                   0x00007fff5b3f5e81 _dispatch_continuation_pop + 472
21  libdispatch.dylib                   0x00007fff5b3ed733 _dispatch_async_redirect_invoke + 703
22  libdispatch.dylib                   0x00007fff5b3e49a9 _dispatch_root_queue_drain + 515
23  libdispatch.dylib                   0x00007fff5b3e4755 _dispatch_worker_thread3 + 101
24  libsystem_pthread.dylib             0x00007fff5b734169 _pthread_wqthread + 1387
25  libsystem_pthread.dylib             0x00007fff5b733be9 start_wqthread + 13

Thread 0:
0   libsystem_kernel.dylib              0x00007fff5b56320a mach_msg_trap + 10
1   com.apple.CoreFoundation            0x00007fff335dc785 __CFRunLoopServiceMachPort + 341
2   com.apple.CoreFoundation            0x00007fff335dbad7 __CFRunLoopRun + 1783
3   com.apple.CoreFoundation            0x00007fff335db153 CFRunLoopRunSpecific + 483
4   com.apple.Foundation                0x00007fff356d7f26 -[NSRunLoop(NSRunLoop) runMode:beforeDate:] + 277
5   com.devmate.DevMateKit              0x000000010273784a -[DMActivationController showActivationWindowIfNeeds] + 1365
6   com.devmate.DevMateKit              0x0000000102736ffc -[DMActivationController updateStepControllerForCurrentStep] + 4480
7   com.devmate.DevMateKit              0x0000000102732f9e -[DMActivationController runActivationWindowInMode:initialActivationInfo:withCompletionHandler:] + 718
8   com.devmate.DevMateKit              0x00000001027312b4 -[DMTrialController beginReminderTrial] + 507
9   com.devmate.DevMateKit              0x0000000102730f9c __36-[DMTrialController beginStartTrial]_block_invoke_2 + 480
10  com.devmate.DevMateKit              0x0000000102730ca4 -[DMTrialController beginStartTrial] + 726
```

**Comment by mpaddle** [comment]
Hi @rgbworld, thank you for reporting this issue. We appreciate the level of detail you've provided.

We've been able to replicate the immediate issue and can confirm that it's an SDK bug. The bug won't affect just your scenario so we'll get look into getting the issue resolved ASAP, likely as part of the next release.

**Comment by GregularExpressions** [comment]
Hey @rgbworld I believe this issue is resolved as of Friday's release [v4.0.1](https://github.com/PaddleHQ/Mac-Framework-V4/releases/tag/v4.0.1) - let us know if you experience any further problems.

Thanks,
Greg.
